ui_framework/doc_site/src/views/pagination/pagination_layouts.js

```diff
@@ -3,22 +3,90 @@ import React, {
 } from 'react';
 
 import {
+  KuiButtonEmpty,
+  KuiContextMenu,
+  KuiPopover,
   KuiPagination,
   KuiFlexGroup,
   KuiFlexItem,
   KuiHorizontalRule,
-  KuiText,
 } from '../../../../components';
 
+function convertPanelTreeToMap(panel, map = {}) {
+  if (panel) {
+    map[panel.id] = panel;
+
+    if (panel.items) {
+      panel.items.forEach(item => convertPanelTreeToMap(item.panel, map));
+    }
+  }
+
+  return map;
+}
+
+function extractPreviousIds(panels) {
+  const idToPreviousPanelIdMap = {};
+
+  Object.keys(panels).forEach(panelId => {
+    const panel = panels[panelId];
+    if (Array.isArray(panel.items)) {
+      panel.items.forEach(item => {
+        const isCloseable = Boolean(item.panel);
+        if (isCloseable) {
+          idToPreviousPanelIdMap[item.panel.id] = panel.id;
+        }
+      });
+    }
+  });
+
+  return idToPreviousPanelIdMap;
+}
+
 export default class extends Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      activePage: 0
+      isPopoverOpen: false,
+      activePage: 0,
+    };
+
+    const panelTree = {
+      id: 0,
+      items: [{
+        name: '10 rows',
+        icon: 'empty',
```

Comment:
This is a cheat, but I think one we'll likely need over and over again. Sometimes you want to line up a row of items, and only show an icon (say a check) on the active row. While you could toggle classes to achieve the same thing, it's much easier (and scalable) to just have an "empty" icon which takes up the space of an icon, but contains no real content.

![image](https://user-images.githubusercontent.com/324519/31001801-464ce464-a49b-11e7-9974-75b2b7cf2e06.png)